Caltech-IPAC/firefly
# Firefly 1314 multi sprectrum table

https://jira.ipac.caltech.edu/browse/FIREFLY-1314

This PR is not connected, yet.  It'll be a bit hard to test.
If you can imagine a piece of code, recognizes a table is a MultiSpectrum ArrayData table, it can forward it to this SearchProcessor.  This processor will be able to expand the table into a meta table with datalinks to all of the spectra in the table.

#### To test:
https://fireflydev.ipac.caltech.edu/firefly-1314-multi-sprectrum-table/firefly/demo/table-api.html
-> click on URL
-> enter URL for this file:  https://github.com/IPAC-SW/irsa-spherex-prototype-data/blob/main/examples/ff-result.vot
Unfortunately, it's a private repo.  You can't access it directly.  One option is to copy it to your webapps/ROOT/ directory and serve it from your host machine. e.g. `http://134.4.107.172:8080/ff-result.vot`
-> Use ProcID=multi_spectrum
-> Show table

You will notice that it's a data product table with the required columns at the end of the table.
Open the access_url with your browser, you'll see a links table with 2 rows, one for each spectrum.  
Follow the accuss_url to see the expanded spectrum table.  
NOTE:  Because it's XML, you have to decode the URL to have it working.

This is a draft.  We need to drop this into an app that know how to use it.

**Update- 10/25** _(from Trey)_: add support for file analysis to recognize multi spectrum. Now this is testable using the upload panel

#### Upload testing:
   - https://fireflydev.ipac.caltech.edu/firefly-1314-multi-sprectrum-table/firefly
   - dowload the votable to you load machine
   - in firefly upload the file
